lib/marshal/marshal.c

```diff
@@ -0,0 +1,185 @@
+                
+                //convert data to this format:
+                //<command name> <key> <flags> ...
+                marshaled_value = calloc(1, size);
+                final_marshed_value = marshaled_value;
+                snprintf(marshaled_value, COMMAND_LENGTH,"%s", "set");
+                marshaled_value += COMMAND_LENGTH; 
+                snprintf(marshaled_value, SPACE_LENGTH ,"%s", space);
+                marshaled_value += SPACE_LENGTH; 
+                snprintf(marshaled_value + data->key_length,"%s", data->key);
+                marshaled_value += data->key_length; 
+                snprintf(marshaled_value, SPACE_LENGTH ,"%s", space);
+                marshaled_value += SPACE_LENGTH; 
+                snprintf(marshaled_value, FLAG_LENGTH ,"%d", 0);
+                marshaled_value += FLAG_LENGTH; 
+                snprintf(marshaled_value, SPACE_LENGTH ,"%s", space);
+                marshaled_value += SPACE_LENGTH; 
+                snprintf(marshaled_value, EXP_TIME_LENGTH ,"%d", 0);
+                marshaled_value += EXP_TIME_LENGTH; 
+                snprintf(marshaled_value, SPACE_LENGTH ,"%s", space);
+                marshaled_value += SPACE_LENGTH; 
+                
+                
+                snprintf(marshaled_value, strlen(value_length) ,"%s", value_length);
+                marshaled_value += strlen(value_length); 
+                snprintf(marshaled_value, SPACE_LENGTH ,"%s", space);
+                marshaled_value += SPACE_LENGTH; 
+                snprintf(marshaled_value, NEWLINE_LENGTH ,"%s", "\r\n");
+                marshaled_value += NEWLINE_LENGTH; 
+                //snprintf(marshaled_value, SPACE_LENGTH ,"%s", space);
+                //marshaled_value += SPACE_LENGTH; 
+                snprintf(marshaled_value, data->value_length ,"%d", data->value_length);
+                marshaled_value += data->value_length; 
+                snprintf(marshaled_value, NEWLINE_LENGTH ,"%s", "\r\n");
+              
+                break;
+            
+            case GET:
+                //convert data to this format
+                // get <key>* \r\n
+                size = COMMAND_LENGTH + SPACE_LENGTH + NEWLINE_LENGTH +
+                       data->key_length ;
+
+                marshaled_value = calloc(1, size);
+                snprintf(marshaled_value, COMMAND_LENGTH,"%s", "get");
+                marshaled_value += COMMAND_LENGTH; 
+                snprintf(marshaled_value, SPACE_LENGTH ,"%s", space);
+                marshaled_value += SPACE_LENGTH; 
+                snprintf(marshaled_value, data->key_length,"%s", data->key);
+                marshaled_value += data->key_length; 
+                snprintf(marshaled_value, NEWLINE_LENGTH ,"%s", "\r\n");
+                marshaled_value += NEWLINE_LENGTH; 
+
+                
+                break;
+            default:
+                assert(0);
+        }
+        *ascii = final_marshed_value;
+        return 0;
```

Comment:
Definitely. Since the code is kind of very half way. I am sure developers (including me ) will take care of these comments when submitting the stable code.
@huyang1988 @akorzan @Grace-Liu : Please correct such issues when you work on stabilizing the code.